block/manager.go

```diff
@@ -734,11 +734,14 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 func (m *Manager) submitBlocksToDA(ctx context.Context) error {
```

Comment:
This still needs a test that verifies that blocks that are failed to submit to the DA aren't removed from the pending blocks. 

This test should fail on main, by checking that pending blocks is not empty after calling `submitBlocksToDA`. Then this test would pass on this branch.

Reply:
added a new test case and made sure it fails in the old code, see below where the reset pending blocks removed the newly added block resulting in failure to include in DA:
```
submitting 2
removing 3
submitting 1
removing 1
submitting 2
removing 2
submitting 2
removing 2
submitting 2
removing 2
--- FAIL: TestSubmitBlocksToDA (0.13s)
    loggers.go:42: Test logs can be found: /var/folders/31/2zc1b7451bs1kk332tl49sk80000gn/T/TestSubmitBlocksToDAnode0.log1423205834
    full_node_integration_test.go:390:
        	Error Trace:	/Users/gupadhyaya/go/src/github.com/gupadhyaya/rollkit/node/full_node_integration_test.go:390
        	Error:      	Should be true
        	Test:       	TestSubmitBlocksToDA
        	Messages:   	3
```